homeassistant/components/azure_service_bus/notify.py

```diff
@@ -0,0 +1,83 @@
+"""Support for azure service bus notification."""
+import json
+
+from azure.servicebus.aio import Message, ServiceBusClient
+import voluptuous as vol
+
+from homeassistant.components.notify import (
+    ATTR_DATA,
+    ATTR_TARGET,
+    ATTR_TITLE,
+    PLATFORM_SCHEMA,
+    BaseNotificationService,
+)
+from homeassistant.const import CONTENT_TYPE_JSON
+import homeassistant.helpers.config_validation as cv
+
+CONF_CONNECTION_STRING = "connection_string"
+CONF_QUEUE_NAME = "queue"
+CONF_TOPIC_NAME = "topic"
+
+ATTR_ASB_MESSAGE = "message"
+ATTR_ASB_TITLE = "title"
+ATTR_ASB_TARGET = "target"
+
+PLATFORM_SCHEMA = vol.All(
+    cv.has_at_least_one_key(CONF_QUEUE_NAME, CONF_TOPIC_NAME),
+    PLATFORM_SCHEMA.extend(
+        {
+            vol.Required(CONF_CONNECTION_STRING): cv.string,
+            vol.Exclusive(
+                CONF_QUEUE_NAME, "output", "Can only send to a queue or a topic."
+            ): cv.string,
+            vol.Exclusive(
+                CONF_TOPIC_NAME, "output", "Can only send to a queue or a topic."
+            ): cv.string,
+        }
+    ),
+)
+
+
+def get_service(hass, config, discovery_info=None):
+    """Get the notification service."""
+    connection_string = config[CONF_CONNECTION_STRING]
+    queue_name = config.get(CONF_QUEUE_NAME)
+    topic_name = config.get(CONF_TOPIC_NAME)
+
+    # Library can do synchronous IO when creating the clients.
+    # Passes in loop here, but can't run setup on the event loop.
+    servicebus = ServiceBusClient.from_connection_string(
+        connection_string, loop=hass.loop
+    )
+
+    if queue_name:
+        client = servicebus.get_queue(queue_name)
```

Comment:
We should except the possible exceptions that this can raise, and take appropriate action. Eg if the resource is not found is unrecoverable, right? Connection error could perhaps be recoverable. Can we rule out that the connection error is due to incorrect connection_string?

`ServiceBusConnectionError`
`ServiceBusResourceNotFound`

Reply:
I think both of them is will be raised if the user sends in incorrect information on the connection string. The `.from_connection_string(...)` method above til validate the format and that all the needed parts are included, but won't check if the resources are there before getting the client. For instance if the domain is incorrect it would be raised as `ServiceBusConnectionError`, and would need the user to update the configuration or fix the resource in Azure and give new config. 

Even if I manage to research which inner exceptions would be retryable due to intermittent connection errors, it doesn't seem like raising `PlatformNotReady` work during setup on notification services? I think it might be best to just log the error better than what the general exception handling in notify component does, to inform the user as best as possible, and return `None` to stop the setup. Agree or am I missing something?

Reply:
Sounds good!